### PR TITLE
Added type definitions for javascript-color-gradient

### PR DESCRIPTION
**Pull request by wilsoncwc**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by wilsoncwc** [comment]
@Adrinlol Hi Lasha, I was using your JavaScript Color Gradient library in a Typescript project and created a simple declaration for it. Would you mind reviewing this PR and/or being added as a second Owner?

File: types/javascript-color-gradient/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Type definitions for javascript-color-gradient 1.3
+// Project: https://github.com/Adrinlol/javascript-color-gradient
+// Definitions by: Wilson Chua <https://github.com/wilsoncwc/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = Gradient;
+
+declare class Gradient {
+    constructor(gradients?: string, maxNum?: number, color?: string[], intervals?: any[]);
```
Comment (peterblazejewicz) [inline types/javascript-color-gradient/index.d.ts:9]:
```suggestion
    constructor(gradients?: string, maxNum?: number, colors?: string[], intervals?: any[]);
```

File: types/javascript-color-gradient/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    setGradient(...args: string[]): Gradient;
```
Comment (peterblazejewicz) [inline types/javascript-color-gradient/index.d.ts:10]: should this be:
```ts
setGradient(color1: string, color2: string, ...otherColors: string[]):  this;
```

File: types/javascript-color-gradient/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    getArray(): string[];
+    getColor(index: number): string;
+    setMidpoint(maxNumber: number): Gradient;
```
Comment (peterblazejewicz) [inline types/javascript-color-gradient/index.d.ts:13]:
```suggestion
    setMidpoint(maxNumber: number): this;
```

File: types/javascript-color-gradient/index.d.ts (hunk 2 above)
Comment (wilsoncwc) [reply types/javascript-color-gradient/index.d.ts:10]: Yep that works nicely, thanks!

```suggestion
    setGradient(color1: string, color2: string, ...otherColors: string[]):  this;
```

**Review by andrewbranch — APPROVED** [review]

**Comment by wilsoncwc** [comment]
Ready to merge

**Pull request merged**